### ISSUE TITLE
no-duplicate-destructuring

### ISSUE THREAD

**Issue by hornta** [eslint/eslint#11200]
Report on duplicate or uneccessary destruct assignments

[ ] Warns about a potential error (problem)
[X ] Suggests an alternate way of doing something (suggestion)
[ ] Enforces code style (layout)
[ ] Other (please specify:)

```js
const myObject = { foo: 1, bar: 2 };
const { foo } = myObject;
const { bar } = myObject;

// can be written as a single destruct assignment
const { foo, bar } = myObject;
```

**Are you willing to submit a pull request to implement this rule?**
I do not have the time right now, but later yea

**Possible names:**
no-unneccessary-destructuring
no-duplicate-destructuring
no-extra-destructuring

**Comment by platinumazure** [comment]
Combining statements is always a tricky fix, so I would recommend not implementing autofix for this rule. But I'll support adding this in.

**Comment by hornta** [comment]
I have moved this into an option in https://github.com/eslint/eslint/issues/11215
